### PR TITLE
Spatial plus sign fix

### PR DESCRIPTION
**Pull request by btylerburton**
# Pull Request

Related to https://github.com/GSA/data.gov/issues/3549

## PR TASKS

- [X] The actual code changes.
- [X] Tests written and passed.
- [X] Any changes to docs?
- [X] Bumped version number in [setup.py](https://github.com/GSA/ckanext-geodatagov/blob/main/setup.py#L13) (also checked on [PyPi](https://pypi.org/project/ckanext-geodatagov/#history)).

### CODE REVIEW

File: ckanext/geodatagov/tests/test_update_geo.py [hunk 1]
```diff
@@ -101,17 +108,25 @@ def test_create_sitemap(self):
         keys = list(json.loads(extras['spatial']).keys())
         assert 'coordinates' in keys
 
-        # this dataset don't have any spatial data
+        # This dataset already include good spatial data
```
Comment (nickumia-reisys) [inline ckanext/geodatagov/tests/test_update_geo.py:111]:
```suggestion
        # This dataset already includes good spatial data
```

File: ckanext/geodatagov/logic.py [hunk 2]
```diff
@@ -459,6 +459,9 @@ def translate_spatial(old_spatial):
                    '"coordinates": [[[{minx}, {miny}], [{minx}, {maxy}], '
                    '[{maxx}, {maxy}], [{maxx}, {miny}], [{minx}, {miny}]]]}}')
 
+    # replace all instances of '+' as this creates bad JSON https://github.com/GSA/data.gov/issues/3549
+    old_spatial = old_spatial.replace('+', '')
+
```
Comment (jbrown-xentity) [inline ckanext/geodatagov/logic.py:464]: I think we want to try to leave `old_spatial` as is, as that is the value given to us by the provider. I would rather see this change down near line 480, as the start of the transformation we do in that specific use case.

**Review by jbrown-xentity — COMMENTED** [review]
Thanks for working on this @btylerburton , one small change I think!

**Comment by nickumia-reisys** [comment]
> > How does removing the `+` make this work now? What about `-` signs?
> 
> We found that our code wasn't handling `+` signs correctly, although the `-` are required for negative lat/long values. The `+` is pretty atypical (and is definitely not a part of the geojson spec, which expects an actual JSON number type), but it's way easier for us to handle this use case than to try to get the data providers to change how they are providing the data...

That makes sense.  You're saying that whatever library couldn't convert the string `"+23"` to the number `23`.  Okay.. There weren't any other issues with format parsing?  It seems like this fix is one that should be handled differently.  I think the problem stems from there not being a spec about this.  But it should be a validation error and if we want to allow it anyway, there should be a connection between the validation failing and it entering this logic.  But that's definitely out of the scope of this effort...

File: ckanext/geodatagov/logic.py (hunk 2 above)
Comment (nickumia-reisys) [reply ckanext/geodatagov/logic.py:464]: I agree with @jbrown-xentity (having more context on this now)

Comment (btylerburton) [reply ckanext/geodatagov/logic.py:464]: I had chained the fix into line 481 `old_spatial.strip().split(',')` but then I realized that old_spatial is getting loaded in at [L467](https://github.com/GSA/ckanext-geodatagov/blob/spatial-plus-sign-fix/ckanext/geodatagov/logic.py#L467) too and that we need to transform it before that.

Comment (btylerburton) [reply ckanext/geodatagov/logic.py:464]: Is anyone clear where initial `spatial` is getting saved into `old_spatial`? If I can find that I can ensure the initial value gets saved unchanged.

File: ckanext/geodatagov/logic.py [hunk 3]
```diff
@@ -459,6 +459,9 @@ def translate_spatial(old_spatial):
...
     # Analyze with type of data is JSON valid
     try:
         geometry = json.loads(old_spatial)  # NOQA F841
```
Comment (nickumia-reisys) [inline ckanext/geodatagov/logic.py:467]:
```suggestion
    # replace all instances of '+' as this creates bad JSON https://github.com/GSA/data.gov/issues/3549
    old_spatial_loadable = old_spatial.replace('+', '')

    # Analyze with type of data is JSON valid
    try:
        geometry = json.loads(old_spatial_loadable)  # NOQA F841
```

**Comment by btylerburton** [comment]
Some further edification from the spec: https://stackoverflow.com/questions/26667313/does-json-allow-positive-sign-for-numbers

**Comment by nickumia-reisys** [comment]
[For clarity](https://github.com/GSA/ckanext-geodatagov/blob/7e317d2ec82c24a4728592ab05243f05bcf0d611/ckanext/geodatagov/logic.py#L424), the internal variable in the `translate_spatial` function was never actually modifying the true "old_spatial" variable.  The change of naming helps to highlight that, so overall, great reviewing, but there was no functionality at stake.

**Pull request merged**